src/webui/webapplication.cpp

```diff
@@ -486,6 +486,24 @@ Http::Response WebApplication::processRequest(const Http::Request &request, cons
     m_currentSession = nullptr;
     m_request = request;
     m_env = env;
+    m_params.clear();
+    if (m_request.method == Http::METHOD_GET) {
+        // Parse GET parameters
+        using namespace Utils::ByteArray;
+        for (const QByteArray &param : copyAsConst(splitToViews(m_request.query, "&"))) {
+            const int sepPos = param.indexOf('=');
+            if (sepPos > 0) { // ignores params without value
+                const QString paramName {QString::fromUtf8(param.constData(), sepPos)};
+                const int valuePos = sepPos + 1;
+                const QString paramValue {
+                    QString::fromUtf8(param.constData() + valuePos, param.size() - valuePos)};
```

Comment:
if `param = "x="` then paramValue is empty, should you filter it out?

Reply:
and you have 2 similar (if not the same) instances, I suppose you want to make it a helper/common function ?

Reply:
Do you mean parameter parsing?

Reply:
>Can the differences be unified?

I do not think it is easy to do without excessive data copying (and, as a consequence, memory allocation). If you want to practice that, you're welcome.